### ISSUE TITLE
iOS: <React/RCTComponent.h> not found.

### ISSUE THREAD

**Issue by jkomyno** [invertase/react-native-firebase#199]
- react-native version: 0.44.3, clean install
- react-native-firebase version: 2.0.0

I have followed the official guide, installed the pods required, opened the .xcworkspace file instead of .xcodeproj (like noted in bold in the guide), but yet I can't manage to get a clean compilation. If I remove react-native-firebase I get I successful build, so I'm sure this issue is related to this particular library. I've also added the plist file downloaded from the Google Console and edited the AppDelegate.m file accordingly, so everything should be set up fine. I just need I kind guidance to solve the matter.

In short: how can I be able to make React (which is in the Project target) visible to RNFirebase (which is in the Pods target)?

**Comment by zenyr** [comment]
I was able to circumvent the issue by sticking with the guide, by adding `Yoga` and `React` in the Podfile manually.

```
  pod "Yoga", :path => "../node_modules/react-native/ReactCommon/yoga"
  pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'BatchedBridge', # Required For React Native 0.45.0+
    'Core',
    # Add any other subspecs you want to use in your project
  ]
```

However I am facing a different issue (`FIRMessaging` looks like outdated, missing `shouldEstablishDirectChannel` method and not deprecating `disconnect`..? ) so I am not 100% sure yet.

**Comment by Salakar** [comment]
@zenyr check your `podfile.lock` - make sure firebase is upgraded to v4+, see our test app `podfile.lock` for reference: https://github.com/invertase/react-native-firebase/blob/master/tests/ios/Podfile.lock (don't copy the whole file directly though - the directory structure on tests is slightly different from a normal project)

**Comment by Salakar** [comment]
Seems to be a few issues with files not found though since the latest react / rnfirebase v2 - on of the causes yesterday was a bug in npm v5. Not sure on the rest yet though, we've gone through the guide and got it working several times - the test app in `/tests` is also running RN v44 and latest pods etc - so not sure at this point what's going on.

Guess we could push up a starter repo with it all setup if that helps anyone

**Comment by jkomyno** [comment]
Thanks a lot @zenyr, your tip helped me managing to build successfully. Now the project runs without issues. @Salakar, I guess that instead on spending a lot of time writing a starter repo for every major (and recent) react native versions (say 0.43, 0.44, 0.45), it would be better to add @zenyr suggestion to the official README, or maybe to link issues of this kind to a bried FAQ section. Whatever. Thanks folks
